Tools/clinic/clinic.py

```diff
@@ -5683,35 +5671,39 @@ def add_parameter(text: str) -> None:
         if not f.docstring_only:
             add("\n" + sig_end_marker + "\n")
 
-        docstring_first_line = output()
+        signature_line = output()
 
         # now fix up the places where the brackets look wrong
-        docstring_first_line = docstring_first_line.replace(', ]', ',] ')
+        return signature_line.replace(', ]', ',] ')
 
-        # okay.  now we're officially building the "parameters" section.
-        # create substitution text for {parameters}
+    @staticmethod
+    def format_docstring_parameters(params: list[Parameter]) -> str:
+        """Create substitution text for {parameters}"""
+        text, add, output = _text_accumulator()
         spacer_line = False
-        for p in parameters:
-            if not p.docstring.strip():
+        for param in params:
+            docstring = param.docstring.strip()
+            if not docstring:
                 continue
             if spacer_line:
                 add('\n')
             else:
                 spacer_line = True
             add("  ")
-            add(p.name)
+            add(param.name)
             add('\n')
-            add(textwrap.indent(rstrip_lines(p.docstring.rstrip()), "    "))
-        parameters_output = output()
-        if parameters_output:
-            parameters_output += '\n'
-
-        ##
-        ## docstring body
-        ##
+            stripped = rstrip_lines(docstring.rstrip())
```

Comment:
You've already called ``docstring = param.docstring.strip()`` above, so the ``str.rstrip`` here is redundant, I believe. (There's a potential behaviour change in that previously it looks like leading whitespace from ``p.docstring`` was kept, but that's likely inconsequential.)

Reply:
If there's a behaviour change, this PR will blow up because it will end up touching all docstrings in Modules ;)

Reply:
I tried removing all redundant strips and rstrips, but I might have missed some. Thanks for the heads-up.